baybe/parameters/substance.py

```diff
@@ -63,6 +62,12 @@ class SubstanceParameter(DiscreteParameter):
     )
     # See base class.
 
+    kwargs_fingerprint: dict = field(factory=dict)
```

Comment:
you would need to gather all possible accepted kwarg types and also prey that they dont add / change it in the external package

Reply:
Agree with Martin

Reply:
I've refactored the fields in one commit, including a generic type hint `dict[str, Any]`. This has two advantages:
* keys are specific (in fact, these *are* strings, since they are converted to kwargs later)
* `Any` is still better than omitting, even in a completely general context like `x: Any`, because it shows that the type is arbitrary and not that people simply forgot to add an annotation